### PR TITLE
fix(es/compat): Handle `__proto__` edge case in `shorthand` pass

### PR DESCRIPTION
**Pull request by magic-akari**
**Related issue:**

 - https://github.com/babel/babel/pull/12664

### CODE REVIEW

**Comment by kdy1** [comment]
CI will fail, but some of them are not related to this PR and will be fixed by https://github.com/swc-project/swc/pull/8079

File: crates/swc_ecma_transforms_compat/src/es2015/shorthand_property.rs [hunk 1]
```diff
@@ -62,15 +62,47 @@ impl VisitMut for Shorthand {
...
 
         match prop {
-            Prop::Shorthand(Ident { sym, span, .. }) => {
+            Prop::Shorthand(ident) => {
+                let value = ident.clone().into();
+
                 *prop = Prop::KeyValue(KeyValueProp {
-                    key: PropName::Ident(quote_ident!(*span, sym.clone())),
-                    value: Box::new(quote_ident!(*span, sym.clone()).into()),
+                    key: if ident.sym == js_word!("__proto__") {
```
Comment (kdy1) [inline crates/swc_ecma_transforms_compat/src/es2015/shorthand_property.rs:69]: Can you rebase?

Comment (magic-akari) [reply crates/swc_ecma_transforms_compat/src/es2015/shorthand_property.rs]: done

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_ecma_transforms_compat

**Pull request merged**